lib/utils/docker.coffee

```diff
@@ -244,3 +290,72 @@ exports.getDocker = (options) ->
 exports.getImageSize = (docker, image) ->
 	docker.getImage(image).inspectAsync()
 	.get('Size')
+
+hasQemu = ->
+	fs = require('mz/fs')
+
+	getQemuPath()
+	.then (qemu) ->
+		fs.stat(qemu)
+	.then ->
+		return true
+	.catch ->
+		return false
+
+getQemuPath = ->
+	resin = require('resin-sdk-preconfigured')
+	path = require('path')
+	fs = require('mz/fs')
+
+	resin.settings.get('binDirectory')
+	.then (binDir) ->
+		# The directory might not be created already,
+		# if not, create it
+		fs.access(binDir)
+		.catch (e) ->
+			if e.code == 'ENOENT'
+				fs.mkdir(binDir)
+			else
+				throw e
+		.then ->
+			path.join(binDir, 'qemu-execve')
+
+platformNeedsQemu = ->
+	os = require('os')
+	os.platform() == 'linux'
+
+installQemu = ->
+	request = require('request')
+	fs = require('fs')
+	zlib = require('zlib')
+
+	getQemuPath()
+	.then (qemuPath) ->
+		installStream = fs.createWriteStream(qemuPath)
+
+		qemuUrl = 'https://github.com/resin-io/qemu/releases/download/v2.5.50-resin-execve/qemu-execve.gz'
```

Comment:
It might be nicer to separate the version out of this string to make it more obvious where we can change it?